### PR TITLE
CMSIS-NN: Update DW conv to new interface

### PR DESCRIPTION
**Pull request by felix-johnny**
1. The DW conv functions are updated as per the new interface
    using structs.
2. New wrapper file/function arm_depthwise_conv_wrapper_s8 is added

### CODE REVIEW

File: CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c [hunk 1]
```diff
@@ -49,71 +49,75 @@
...
-                                     const int32_t input_ch,
-                                     const int8_t *kernel,
-                                     const int32_t output_ch,
-                                     const int32_t pad_x,
-                                     const int32_t pad_y,
-                                     const int32_t stride_x,
-                                     const int32_t stride_y,
+
+arm_status arm_depthwise_conv_3x3_s8(const cmsis_nn_context *ctx,
+                                     const cmsis_nn_conv_params *conv_params,
```
Comment (gmiodice) [inline CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c:54]: What do you think about having cmsis_nn_dwc_params which includes the depth multiplier as well?

Comment (gmiodice) [reply CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c:54]: This should also help integrating future supports as well

File: CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_wrapper_s8.c [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+#include "arm_math.h"
+#include "arm_nnfunctions.h"
+#include "arm_nnsupportfunctions.h"
+
+/**
+ *  @ingroup groupNN
+ */
+
+/**
+ * @addtogroup NNConv
```
Comment (gmiodice) [inline CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_wrapper_s8.c:40]: Not sure, do you have the tag also for depthwise?

File: CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c (hunk 1 above)
Comment (felix-johnny) [reply CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c:54]: Good point. Lets do that!

File: CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_wrapper_s8.c (hunk 2 above)
Comment (felix-johnny) [reply CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_wrapper_s8.c:40]: no..its all bundled into conv..

File: CMSIS/NN/Include/arm_nn_types.h [hunk 3]
```diff
@@ -88,6 +88,19 @@ typedef struct
...
+typedef struct
+{
+    int32_t             input_offset;   /**< Zero value for the input tensor */
+    int32_t             output_offset;  /**< Zero value for the output tensor */
+    int32_t             ch_mult;        /**< Channel Multiplier. ch_mult * in_ch = out_ch */
+    cmsis_nn_tile       stride;
+    cmsis_nn_tile       padding;
+    cmsis_nn_tile       dilation;
+    cmsis_nn_activation activation;
+
```
Comment (gmiodice) [inline CMSIS/NN/Include/arm_nn_types.h:101]: Remove empty line?

File: CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c [hunk 4]
```diff
@@ -49,71 +49,74 @@
...
-                                     const int32_t pad_x,
-                                     const int32_t pad_y,
-                                     const int32_t stride_x,
-                                     const int32_t stride_y,
+
+arm_status arm_depthwise_conv_3x3_s8(const cmsis_nn_context *ctx,
+                                     const cmsis_nn_dw_conv_params *dw_conv_params,
+                                     const cmsis_nn_per_channel_quant_params *quant_params,
+                                     const cmsis_nn_dims *input_dims,
+                                     const q7_t *input,
```
Comment (gmiodice) [inline CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c:57]: input_data?

File: CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c [hunk 5]
```diff
@@ -49,71 +49,74 @@
...
+                                     const cmsis_nn_dims *filter_dims,
+                                     const q7_t *kernel,
```
Comment (gmiodice) [inline CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c:59]: filter_data?

File: CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c [hunk 6]
```diff
@@ -49,71 +49,74 @@
...
+                                     const cmsis_nn_dims *bias_dims,
                                      const int32_t *bias,
```
Comment (gmiodice) [inline CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c:61]: bias_data?

File: CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c [hunk 7]
```diff
@@ -49,71 +49,74 @@
...
-                                     int8_t *output,
```
Comment (gmiodice) [inline CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c:63]: output_data?

File: CMSIS/NN/Include/arm_nn_types.h (hunk 3 above)
Comment (felix-johnny) [reply CMSIS/NN/Include/arm_nn_types.h:101]: i'll fix this..

File: CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c (hunk 7 above)
Comment (felix-johnny) [reply CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_3x3_s8.c:63]: for the _data comments, I think it is ok without the suffix as it is still clear that it is input/output/kernel data.

File: CMSIS/NN/Include/arm_nn_types.h (hunk 3 above)
Comment (felix-johnny) [reply CMSIS/NN/Include/arm_nn_types.h]: commented below

**Pull request merged**